source/code/go/src/plugins/oms.go

```diff
@@ -0,0 +1,263 @@
+type DataItem struct {
+	LogEntry          string `json:"LogEntry"`
+	LogEntrySource    string `json:"LogEntrySource"`
+	LogEntryTimeStamp string `json:"LogEntryTimeStamp"`
+	ID                string `json:"Id"`
+	Image             string `json:"Image"`
+	Name              string `json:"Name"`
+	SourceSystem      string `json:"SourceSystem"`
+	Computer          string `json:"Computer"`
+}
+
+// ContainerLogBlob represents the object corresponding to the payload that is sent to the ODS end point
+type ContainerLogBlob struct {
+	DataType  string     `json:"DataType"`
+	IPName    string     `json:"IPName"`
+	DataItems []DataItem `json:"DataItems"`
+}
+
+func populateMaps() {
+	files, err := ioutil.ReadDir(containerInventoryPath)
+
+	if err != nil {
+		Log("error when reading container inventory")
+		log.Fatal(err.Error())
+	}
+
+	for _, file := range files {
+		fullPath := fmt.Sprintf("%s/%s", containerInventoryPath, file.Name())
+		fileContent, err := ioutil.ReadFile(fullPath)
+		if err != nil {
+			Log("Error reading file content %s", fullPath)
+			log.Fatal(err)
+		}
+		var containerInventory ContainerInventory
+		unmarshallErr := json.Unmarshal(fileContent, &containerInventory)
+
+		if unmarshallErr != nil {
+			Log("Unmarshall error when reading file %s", fullPath)
+			log.Fatal(unmarshallErr)
+		}
+
+		ImageIDMap[file.Name()] = containerInventory.Image
+		NameIDMap[file.Name()] = containerInventory.ElementName
+	}
+}
+
+func createLogger() *log.Logger {
+	logfile, err := os.Create(filepath.Join("/shared/data", fmt.Sprintf("fluent-bit-runtime_%s.log", time.Now().Format("2006-01-02T15-04-05"))))
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return log.New(logfile, "", 0)
+}
+
+func initMaps() {
+	ImageIDMap = make(map[string]string)
+	NameIDMap = make(map[string]string)
+
+	populateMaps()
+
+	for range time.Tick(time.Second * 60) {
+		populateMaps()
+	}
+}
+
+func updateIgnoreContainerIds() {
+	IgnoreIDSet = make(map[string]bool)
+
+	updateKubeSystemContainerIDs()
+
+	for range time.Tick(time.Second * 300) {
+		updateKubeSystemContainerIDs()
+	}
+}
+
+func updateKubeSystemContainerIDs() {
+
+	if strings.Compare(os.Getenv("DISABLE_KUBE_SYSTEM_LOG_COLLECTION"), "true") != 0 {
+		Log("Kube System Log Collection is ENABLED.")
+		return
+	}
+
+	Log("Kube System Log Collection is DISABLED. Collecting containerIds to drop their records")
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		Log("Error getting config")
+		panic(err.Error())
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		Log("Error getting clientset")
+		panic(err.Error())
+	}
+
+	pods, err := clientset.CoreV1().Pods("kube-system").List(metav1.ListOptions{})
+	if err != nil {
+		Log("Error getting pods")
+		panic(err.Error())
+	}
+
+	for _, pod := range pods.Items {
+		for _, status := range pod.Status.ContainerStatuses {
+			IgnoreIDSet[status.ContainerID[9:len(status.ContainerID)]] = true

```

Comment:
So are we leaking entr(ies) for older containers that doesn't exist as we don't clear entry for non existing containers? (Same in other maps as well).